packages/smart-wallet/src/smartWallet.js

```diff
@@ -104,8 +98,9 @@ const mapToRecord = map => Object.fromEntries(map.entries());
  * @typedef {{
  *   agoricNames: ERef<import('@agoric/vats').NameHub>,
  *   registry: {
- *     getRegisteredAsset: (b: Brand) => BrandDescriptor,
- *     getRegisteredBrands: () => BrandDescriptor[],
+ *     has: (b: Brand) => boolean,
+ *     get: (b: Brand) => BrandDescriptor,
+ *     values: () => BrandDescriptor[],
```

Comment:
this looks a lot like `Map`. consider,
```suggestion
 *     Pick<Map<Brand, BrandDescription>, 'has' | 'get' | 'values'
```
Even better, define `AssetRegistry` type in `walletFactory.js` where the behavior is defined and import that type here.

Reply:
I think of the contract / interface as defined by the consumer, and walletFactory.js provides a suitable implementation.

I like the `Pick<...>` suggestion, but `Map.values` returns an iterator, not an array... this whole `Far` facet thing isn't durable, though, is it?

Reply:
um... what's the difference between `Iterable` and `IterableIterator`? why doesn't the `@agoric/store` type match the standard JS/TS type?

```
  The types returned by 'registry.values()' are incompatible between these types.
    Property 'next' is missing in type 'Iterable<BrandDescriptor>' but required in type 'IterableIterator<BrandDescriptor>'
```